scripts/linters/pylint_extensions.py

```diff
@@ -1654,99 +1654,110 @@ class SingleLineCommentChecker(checkers.BaseChecker):
-
-            # Single multi-line comment, ignore it.
-            if line.count(multi_line_indicator) == 2:
-                continue
-
-            # Flip multi-line boolean depending on whether or not we see
-            # the multi-line indicator. Possible for multiline comment to
-            # be somewhere other than the start of a line (e.g. func arg),
-            # so we can't look at start of or end of a line, which is why
-            # the case where two indicators in a single line is handled
-            # separately (i.e. one line comment with multi-line strings).
-            if multi_line_indicator in line:
-                in_multi_line_comment = not in_multi_line_comment
-
-            # Ignore anything inside a multiline comment.
-            if in_multi_line_comment:
-                continue
+        Returns:
+            bool. False if there is no space at the beginning of the comment
+            else returns True.
+        """
+        if re.search(br'^#[^\s].*$', line) and not line.startswith(b'#!'):
+            self.add_message(
+                'no-space-at-beginning', line=line_num)
+            return False
+        return True
 
-            next_line = ''
-            previous_line = ''
-            if line_num + 1 < file_length:
-                next_line = file_content[line_num + 1].strip()
-            if line_num > 0:
-                previous_line = file_content[line_num - 1].strip()
+    def _check_no_capital_letter_at_beginning(self, line, line_num):
+        """Checks if the comment starts with a capital letter.
 
-            # Ignore lines which are not comments.
-            if not line.startswith(b'#'):
-                continue
+        Args:
+            line: str. The current line of comment.
+            line_num: int. Line number of the current comment.
+        """
+        # Check if comment contains any excluded phrase.
+        excluded_phrase_is_present_at_beginning = any(
+            word in line.split()[1] for word in EXCLUDED_PHRASES)
+        if (re.search(br'^# [a-z][A-Za-z]*.*$', line) and not
+                excluded_phrase_is_present_at_beginning):
+            self.add_message(
+                'no-capital-letter-at-beginning', line=line_num)
 
-            # Check if comment contains any excluded phrase.
-            word_is_present_in_excluded_phrases = any(
-                word in line for word in EXCLUDED_PHRASES)
+    def process_tokens(self, tokens):
+        """Custom pylint checker to ensure that comments follow correct style.
 
-            # Comments may include a lowercase character at beginning
-            # or may not use a punctuation at end if it contains a
-            # excluded phrase e.g. "# coding: utf-8".
-            if word_is_present_in_excluded_phrases:
-                continue
+        Args:
+            tokens: Token. Object to access all tokens of a module.
+        """
+        allowed_comment_prefixes = ['int', 'str', 'float', 'bool', 'v']
```

Comment:
Can we move this to the configuration file (.pylintrc)?

Reply:
Can't move this to .pylintrc as that file only contains properties not variables or constants.

Reply:
I think you can do this I've left some hint in the below comment PTAL!